test/index.spec.ts

```diff
@@ -137,452 +137,540 @@ test('encryptor:encrypt & decrypt with wrong password', async ({ page }) => {
-  const encryptedPayload = {
-    data: sampleEncryptedData.data,
-    iv: sampleEncryptedData.iv,
-  };
-  const { salt } = sampleEncryptedData;
-
-  await expect(
-    page.evaluate(
-      async (args) => {
-        const key = await window.encryptor.keyFromPassword(
-          args.wrongPassword,
-          args.salt,
-        );
-        return await window.encryptor.decryptWithKey(
-          key,
-          args.encryptedPayload,
-        );
-      },
-      { encryptedPayload, salt, wrongPassword },
-    ),
-  ).rejects.toThrow('Incorrect password');
-});
-
-test('encryptor:importKey generates valid CryptoKey', async ({ page }) => {
-  const isKey = await page.evaluate(
-    async (args) => {
-      const key = await window.encryptor.importKey(args.SAMPLE_EXPORTED_KEY);
-      return key instanceof CryptoKey;
-    },
-    { SAMPLE_EXPORTED_KEY },
-  );
-  expect(isKey).toBe(true);
-});
-
-test('encryptor:exportKey generates valid CryptoKey string', async ({
-  page,
-}) => {
-  const keyString = await page.evaluate(
-    async (args) => {
-      const key = await window.encryptor.importKey(args.SAMPLE_EXPORTED_KEY);
-      return await window.encryptor.exportKey(key);
-    },
-    { SAMPLE_EXPORTED_KEY },
-  );
-  expect(keyString).toStrictEqual(SAMPLE_EXPORTED_KEY);
-});
-
-test('encryptor:encryptWithDetail and decryptWithDetail provide same data', async ({
-  page,
-}) => {
-  const password = 'a sample passw0rd';
-  const data = { foo: 'data to encrypt' };
-
-  const { vault } = await page.evaluate(
-    async (args) =>
-      await window.encryptor.encryptWithDetail(args.password, args.data),
-    { data, password },
-  );
-
-  const decryptedDetail = await page.evaluate(
-    async (args) =>
-      await window.encryptor.decryptWithDetail(args.password, args.data),
-    { data: vault, password },
-  );
-
-  expect(JSON.stringify(decryptedDetail.vault)).toStrictEqual(
-    JSON.stringify(data),
-  );
-});
-
-test('encryptor:decryptWithKey provide same data when using exported key from encryptWithDetail', async ({
-  page,
-}) => {
-  const password = 'a sample passw0rd';
-  const data = { foo: 'data to encrypt' };
-
-  const { vault, exportedKeyString } = await page.evaluate(
-    async (args) =>
-      await window.encryptor.encryptWithDetail(args.password, args.data),
-    { data, password },
-  );
 
-  // Use the exported key and vault to properly decrypt the data
-  const decryptWithKeyResult = await page.evaluate(
-    async (args) => {
-      const key = await window.encryptor.importKey(args.keyString);
-      return await window.encryptor.decryptWithKey(key, JSON.parse(args.data));
-    },
-    { data: vault, keyString: exportedKeyString },
-  );
+      expect(decryptedData).toStrictEqual(data);
+    });
 
-  expect(JSON.stringify(decryptWithKeyResult)).toStrictEqual(
-    JSON.stringify(data),
-  );
-});
+    test('encryptor:encrypt then decrypt using key derived from wrong password', async ({
```

Comment:
This test doesn't rely on either `sampleEncryptedData` or `oldSampleEncryptedData`. Should it be inside the loop?